util/reggen/reg_block.py

```diff
@@ -57,6 +57,9 @@ def __init__(self, reg_width: int, params: ReggenParams):
         # A list of all write enable names
         self.wennames = []  # type: List[str]
 
+        # Whether this block contains shadowed registers
+        self.shadowed = False
```

Comment:
Could we call this something like `has_shadowed_reg`? Otherwise, it sounds like the entire register block is shadowed if `self.shadowed = True`, which isn't what you mean.

Reply:
yep good call, will change.